src/lib/io/sol-iio.c

```diff
@@ -315,7 +319,52 @@ create_sysfs_trigger(struct sol_iio_device *device)
 }
 
 static bool
-check_trigger(struct sol_iio_device *device)
+create_hrtimer_trigger(struct sol_iio_device *device, const char *trigger_name)
+{
+    int success, id, len, r = false;
+    char *name = NULL, *replace;
+    char path[PATH_MAX];
+
+    if (check_file_existence(CONFIGFS_IIO_TRIGGERS_PATH)) {
+        if (strlen(trigger_name) == 0) {
```

Comment:
I prefer the anonymous "hrtimer:" support, for easy use. writing a null string does not break the trigger detect logic, I add a null checking for code-safe